### PR TITLE
refactoring code by extracting some meaningful methods

### PR DESCRIPTION
**Pull request by freewind**

### CODE REVIEW

File: api/request.go [hunk 1]
```diff
@@ -25,33 +25,49 @@ type BuildRequest struct {
...
-				m["asset_id"] = consensus.BTMAssetID.String()
-			default:
-				asset, err := a.wallet.AssetReg.FindByAlias(alias)
-				if err != nil {
-					return errors.WithDetailf(err, "invalid asset alias %s on action %d", alias, i)
-				}
-				m["asset_id"] = asset.AssetID.String()
-			}
+		err := a.completeMissingAssetId(m, i)
+		if err != nil {
```
Comment (Paladz) [inline api/request.go]: if err := a.completeMissingAssetId(m, i); err != nil {

File: api/request.go [hunk 2]
```diff
@@ -25,33 +25,49 @@ type BuildRequest struct {
...
+			return err
+		}
+
+		err = a.completeMissingAccountId(m, i, ctx)
+		if err != nil {
```
Comment (Paladz) [inline api/request.go]: if err = a.completeMissingAccountId(m, i, ctx); err != nil {

File: api/transact.go [hunk 3]
```diff
@@ -60,7 +60,7 @@ func onlyHaveSpendActions(req *BuildRequest) bool {
 }
 
 func (a *API) buildSingle(ctx context.Context, req *BuildRequest) (*txbuilder.Template, error) {
-	err := a.filterAliases(ctx, req)
+	err := a.completeMissingIds(ctx, req)
 	if err != nil {
```
Comment (Paladz) [inline api/transact.go]: if err := a.completeMissingIds(ctx, req); err != nil {

**Comment by freewind** [comment]
Updated~

**Pull request merged**